mesh/model.c

```diff
@@ -803,15 +803,17 @@ static inline void send_fd(void *data, size_t len, void *user_data)
 }
 
 static void send_fd_dev_key_msg_rcvd(uint8_t ele_idx, uint16_t src,
-					uint16_t app_idx, uint16_t net_idx,
-					uint16_t size, const uint8_t *data,
-					send_callback_t send_callback,
-					void *user_data)
+						uint16_t dst, uint16_t app_idx,
+						uint16_t net_idx, uint16_t size,
+						const uint8_t *data,
+						send_callback_t send_callback,
+						void *user_data)
 {
 	struct fd_msg *msg = fd_msg_new(data, size);
 	msg->flags = (1 << 0) | ((app_idx == APP_IDX_DEV_REMOTE) << 1);
 	msg->element_idx = ele_idx;
 	msg->src_addr = src;
+	msg->dst_addr = dst;
```

Comment:
What for? Isn't element index enough? Device key messages are always unicast, so destination address is always node addr + element index.

Reply:
? This is the address of the Gateway that received the DeviceKey message (BlueZ knows it own address, but AMQP knows nothing about the destination address).

Reply:
See below